src/WebJobs.Script/Extensions/HttpRequestExtensions.cs

```diff
@@ -102,6 +104,38 @@ public static bool IsPlatformInternalRequest(this HttpRequest request, IEnvironm
             return string.Compare(value, "true", StringComparison.OrdinalIgnoreCase) == 0;
         }
 
+        public static bool IsInternalAuthAllowed(this HttpRequest httpRequest)
```

Comment:
The name of the method leads the reader to believe that this is just a check to see if internal auth is allowed, but the method is trying to authenticate the request. Can we use a more descriptive name here?